kimchi/src/circuits/polynomials/foreign_field_add/gadget.rs

```diff
@@ -64,20 +60,30 @@ impl<F: PrimeField> CircuitGate<F> {
     ///      [9n+9]           -> 1 Zero row for bound result
     /// ]
     ///
-    pub fn create_foreign_field_add(start_row: usize, num: usize) -> (usize, Vec<Self>) {
-        // Create multi-range-check gates for $a, b, r, u$
-        // ----------------------------------------------
-        let mut circuit_gates = vec![];
-        let mut next_row = start_row;
-
-        CircuitGate::append_multi_range_check_rows(&mut next_row, &mut circuit_gates); // left input
-        for _ in 0..num {
-            for _ in 0..2 {
-                // right input and result
-                CircuitGate::append_multi_range_check_rows(&mut next_row, &mut circuit_gates);
+    pub fn create_ffadd_chain(
```

Comment:
Nice, but I think you may just want to leave the range-checks always separate since we either have already range checked `a` and `b` and simply want to range check `r`.  Or don't want to range check `r`, but only want to range check `a` and `b`.

Reply:
or maybe just `a` or `b`

Reply:
I guess I mostly created this gadget for test purposes, but I presume we won't be using it a lot in a real use case. For that reason we also have the `create_single_ffadd` gadget, which will be much more convenient.

Reply:
I guess you mean to have a completely separate gadget

Reply:
I just mean have your gadget/gate do the simplest thing.. just the ffAdd gate and witness alone, leaving the range checks out.  This will be simpler code for you and leave maximum flexibility to the user/caller of this gate.  I still propose that you have one test that tests the gate with the (external) range checks, but that's just test code.

Reply:
Done! Now there's a simple chain function and external expansion to include rangechecks for tests